src/Tests/Equinor.Procosys.Library.Query.Tests/GetAllDisciplines/GetAllDisciplinesQueryHandlerTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Equinor.Procosys.Library.Query.Client;
+using Equinor.Procosys.Library.Query.GetAllDisciplines;
+using Microsoft.Extensions.Options;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using ServiceResult;
+
+namespace Equinor.Procosys.Library.Query.Tests.GetAllDisciplines
+{
+    [TestClass]
+    public class GetAllDisciplinesQueryHandlerTests
+    {
+        private const string Plant = "PCS$TESTPLANT";
+        private Mock<IBearerTokenApiClient> _clientMock;
+        private Mock<IOptionsMonitor<MainApiOptions>> _optionsMonitorMock;
+        private GetAllDisciplinesQuery _request;
+
+        [TestInitialize]
+        public void Setup()
+        {
+            var options = new MainApiOptions
+            {
+                ApiVersion = "1",
+                Audience = "Aud",
+                BaseAddress = "http://example.com/"
+            };
+
+            _optionsMonitorMock = new Mock<IOptionsMonitor<MainApiOptions>>();
+            _optionsMonitorMock
+                .Setup(x => x.CurrentValue)
+                .Returns(options);
+
+            var classifications = new List<string>
+            {
+                "ClassA",
+                "ClassB"
+            };
+
+            _request = new GetAllDisciplinesQuery(Plant, classifications);
+
+            var url = $"{options.BaseAddress}Library/Disciplines" +
```

Comment:
Depending of request contains classifications or not, handler should build 2 different urls.
Add setup and test with zero classifications to ensure urls are built different.  Setup in line 51 and 66 should be duplicated and return different list of MainApiDiscipline depending on built url